src/DGtal/shapes/doc/moduleHalfEdgeMesh.dox

```diff
@@ -0,0 +1,650 @@
+*/
+namespace DGtal {
+//----------------------------------------
+/*!
+@page moduleHalfEdgeMesh Half-edge data structure, triangulated surfaces and polygonal surfaces
+@author Jacques-Olivier Lachaud
+
+@since 0.9.4
+
+Part of \ref packageTopology and  \ref packageShapes
+  
+This part of the manual describes how to represent combinatorial
+surfaces, generally embedded in \f$\mathbb{R}^3\f$. The underlying
+combinatorial topological structure is the classical half-edge data
+structure (or doubly connected cell list). We also provide a
+triangulated surface representation that is based on an half-edge data
+structure. A more general polygonal surface is provided and is also
+based on the same half-edge data sructure. In the latter case, faces
+of the surface can be arbitrary simple polygons.
+
+[TOC]
+
+The following programs are related to this documentation:
+testHalfEdgeDataStructure.cpp, testTriangulatedSurface.cpp, viewMarchingCubes.cpp, viewPolygonalMarchingCubes.cpp
+
+
+\section HEM_sec1 The half-edge data structure
+
+An half-edge data structure is a way to represent the topology of a
+combinatorial surface. A combinatoirial surface is a union of
+vertices, edges (a curve bordered by two vertices), faces (a piece of
+surface bordered by a sequence of edges). They are often called
+0-cells, 1-cells and 2-cells respectively. The half-edge data
+structure describes which cells are connected to each other. Its
+principle is to associate two \i half-edges to each edge. Once this is
+done, it is easy to tie cells together by simply indicating for each
+half-edge:
+
+- its next half-edge (along its face)
+- its opposite half-edge (along the edge, that the half-edge associated to the neighboring face)
+- its associated face
+- its associated vertex (here, we choose the "to" vertex if the half-edge is seen as an arc)
+- its associated edge
+
+The classical half-edge data structure is implemented in the class HalfEdgeDataStructure.
+
+@note Large parts of this class are taken from
+https://github.com/yig/halfedge, written by Yotam Gingold.
+ 
+
+\subsection HEM_sec1_1 Creating an half-edge data structure
+
+For now, you only have methods to create an half-edge data structure
+from a set of triangles and edges. For instance, the following code
+builds a half-edge data structure representing two triangles tied
+along one edge:
+
+\code
+  #include "DGtal/topology/HalfEdgeDataStructure.h"
+  ...
+  std::vector< Triangle > triangles( 2 );
+  triangles[0].v = { 0, 1, 2 };
+  triangles[1].v = { 2, 1, 3 };
+
+  HalfEdgeDataStructure mesh;
+  mesh.build( triangles );
+\endcode
+
+All elements within the half-edge data structure are numbered,
+starting from 0. Furthermore, the indices of vertices and triangles
+are the same as the one given at initialization. Unordered edges may
+also be numbered according to your preference, but you need to call
+the other HalfEdgeDataStructure::build method, which takes as input
+triangles and edges.
```

Comment:
don't get it: what does exactly ```build``` ? It would render the edges of the triangles  ?

Reply:
It's just that the first build method only requires triangles. So it defines some order for edges. If you wish to force the numbering of edges, you can use the other build method (where the numbering of edges is specified).